gtsam/nonlinear/NonlinearFactorGraph.h

```diff
@@ -202,6 +203,29 @@ namespace gtsam {
       push_back(boost::make_shared<ExpressionFactor<T> >(R, z, h));
     }
 
+    /**
+     * Directly add a PriorFactor to the factor graph.
+     * @param key    Variable key
+     * @param prior  The variable's prior value
+     * @param model  Noise model for prior factor
+     */
+    template<typename T>
+    void addPrior(Key key, const T& prior,
+                  const SharedNoiseModel& model = nullptr) {
+      push_back(boost::make_shared<PriorFactor<T>>(key, prior, model));
+    }
+
+    /**
+     * Directly add a PriorFactor to the factor graph.
+     * @param key         Variable key
+     * @param prior       The variable's prior value
+     * @param covariance  Covariance matrix.
+     */
```

Comment:
Document that smart noise model will pick right noise model (e.g. diagonal).

Reply:
Done 👍